core/server/proxy/src/main/java/alluxio/proxy/s3/S3RestServiceHandler.java

```diff
@@ -178,4 +232,23 @@ private void checkBucketIsAlluxioDirectory(String bucketPath) throws S3Exception
       throw toBucketS3Exception(e, bucketPath);
     }
   }
+
+  /**
+   * Checks the given object is either a file or an empty directory in Alluxio.
+   *
+   * @param objectPath the object path in Alluxio
+   * @throws S3Exception if the object is not a valid in Alluxio
+   */
+  private void checkObjectInAlluxio(String objectPath) throws S3Exception {
```

Comment:
What guarantee do we get by having this method? Invocations to this method do not guarantee future invocations with the same path exist or do not exist.